tools/publish-automation/publish.go

```diff
@@ -0,0 +1,269 @@
+
+	url := "https://github.com/fleetdm/fleet/releases/edit/fleet-" + version
+	err := openURL(url)
+	if err != nil {
+		fmt.Println("Error opening URL:", err)
+		return false
+	}
+
+	fmt.Println("Hit ENTER for next step")
+	fmt.Scanln()
+	return true
+}
+
+func openURL(url string) error {
+	var cmd *exec.Cmd
+
+	switch runtime.GOOS {
+	case "windows":
+		cmd = exec.Command("cmd", "/c", "start", url)
+	case "darwin":
+		cmd = exec.Command("open", url)
+	default:
+		cmd = exec.Command("xdg-open", url)
+	}
+
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func npmPublish() bool {
+	fmt.Println("\n===========================================================")
+	fmt.Println("Step 4: NPM PUBLISH")
+	fmt.Println("Hitting ENTER will take you to another terminal to fleet/tools/fleetctl-npm/")
+	fmt.Println("Once there type \"npm publish\". You will need an auth code for that.")
+	fmt.Scanln()
+	fmt.Println("Taking you to the other terminal ... ")
+
+	relativePath := "./tools/fleetctl-npm"
+	err := openTerminalAtPath(relativePath)
+	if err != nil {
+		fmt.Println("Error opening Terminal:", err)
+		return false
+	}
+
+	fmt.Println("Hit ENTER for next step")
+	fmt.Scanln()
+	return true
+}
+
+func openTerminalAtPath(path string) error {
+	var cmd *exec.Cmd
+
+	switch runtime.GOOS {
+	case "darwin":
+		cmd = exec.Command("open", "-a", "Terminal", path)
+	default:
+		return fmt.Errorf("unsupported operating system")
+	}
+
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkDocker(version string) bool {
+	fmt.Println("\n===========================================================")
+	fmt.Println("Step 5: Check hub.docker for existing version")
+	fmt.Println("Hitting ENTER will take you to the proper hub.docker page.")
+	fmt.Println("Verify that the version exists")
+	fmt.Scanln()
+	url := "https://hub.docker.com/r/fleetdm/fleet/tags?page=1&name=patch-fleet-" + version
+	fmt.Println("Taking you to the docker hub page: " + url)
+
+	err := openURL(url)
+	if err != nil {
+		fmt.Println("Error opening URL:", err)
+		return false
+	}
+
+	fmt.Println("Does Docker version exist? (Y/y)")
+	var answer string
+	fmt.Scanln(&answer)
+	if answer == "Y" || answer == "y" {
+		return true
+	} else {
+		return false
+	}
+}
+
+func deployDogfood(version string) bool {
+	fmt.Println("\n===========================================================")
+	fmt.Println("Step 6: Deploy " + version + " to Dogfood.")
+	fmt.Println("Hitting ENTER will take you to the proper git action page.")
+	fmt.Println("Once at the git action page do this:")
+	fmt.Println("    1 - Go to #help-infrastructure slack channel and paste this:\"@infrastructure-oncall We have just released " + version + " and are now deploying it to dogfood\"  ")
+	fmt.Println("    2 - press the \"Run workflow\" button on the right side of the screen")
+	fmt.Println("    3 - paste \"fleetdm/fleet:patch-fleet-" + version + "\" as input.")
```

Comment:
we have been avoiding using the `patch-fleet` image tag since having to deploy a second time fails if the image hasn't changed. Durring patch releases I always deploy the git tag

Reply:
That's how I release, but if we agree on tags, I don't mind.

Reply:
Using the `patch-fleet-` branch image is good for testing the RC, but at the end we should push the tag image: `fleet-v4.x.x`.

Reply:
https://github.com/fleetdm/fleet/pull/17870 added to publish step of patch_release script